src/hotspot/share/opto/loopnode.cpp

```diff
@@ -3514,9 +3514,9 @@ bool IdealLoopTree::beautify_loops( PhaseIdealLoop *phase ) {
 // encountered.  Helper for check_safepts.
 void IdealLoopTree::allpaths_check_safepts(VectorSet &visited, Node_List &stack) {
```

Comment:
@rwestrel you should update the description here. Suggestion:
```
Allpaths backwards scan. Starting at the head, traversing all backedges, and the body. Terminating each path at first safepoint encountered.  Helper for check_safepts.
```

Reply:
Also the line above is not accurate enough anymore:
`_required_safept->push(n);  // save the one closest to the tail`

For one: could there not be multiple such SafePoints? If so: what does it mean to take "the closest"? And we may have multiple backedges / tails, now that we allow irreducible loops.

Reply:
Other than that the fix looks reasonable to me, thanks for the fix @rwestrel !